documentation/dhcp_subnet.md

```diff
@@ -0,0 +1,93 @@
+| `group`                | String        | Platform dependant               | Group of the generated configuration file                           |                     |
+| `mode`                 | String        | `'0640'`                         | Filemode of the generated configuration file                        |                     |
+| `shared_network`       | True, False   | `false`                          | DHCP failover configuration file path                               |                     |
+| `config_includes_directory` | String   | `/etc/dhcp/dhcpd(6).d`           | Directory to create included configuration files in                 |                     |
+| `lib_dir`              | String        | Platform dependant               | DHCPD lib directory path                                            |                     |
+| `lease_file`           | String        | Platform dependant               | DHCPD lease file path                                               |                     |
+| `allow`                | Array         | `nil`                            | Control DHCPD allowed requests                                      |                     |
+| `deny`                 | Array         | `nil`                            | Control DHCPD denied requests                                       |                     |
+| `ignore`               | Array         | `nil`                            | Control DHCPD ignored requests                                      |                     |
+| `parameters`           | Array, Hash   | `nil`                            | DHCPD global parameters                                             |                     |
+| `options`              | Array, Hash   | `nil`                            | DHCPD global options                                                |                     |
+| `keys`                 | Hash          | `nil`                            | TSIG keys configuration                                             |                     |
+| `zones`                | Hash          | `nil`                            | Dynamic DNS zone configuration                                      |                     |
+| `hooks`                | Hash          | `nil`                            | Server event action configuration                                   |                     |
+| `failover`             | Hash          | `nil`                            | DHCP failover configuration                                         |                     |
+| `include_files`        | Array         | `nil`                            | Additional configuration files to include                           |                     |
+| `extra_lines`          | String, Array | `nil`                            | Extra lines to append to the configuration file                     |                     |
+
+## Examples
+
+```ruby
+dhcp_config '/etc/dhcp/dhcpd.conf' do
```

Comment:
This should be showing a ``dhcp_subnet`` example :)

It might be nice to show both an IPv4 and IPv6 example if possible.

Reply:
Good one, not quite sure how I've managed that!